tmol/database/chemical/__init__.py

```diff
@@ -56,36 +62,80 @@ class Connection:
 
 
 @attr.s(auto_attribs=True, frozen=True, slots=True)
-class ConnectedAtom:
-    atom: str
+class UnresolvedAtom:
+    atom: Optional[str] = None
     connection: Optional[str] = None
+    bond_sep_from_conn: Optional[int] = None
 
 
 @attr.s(auto_attribs=True, frozen=True, slots=True)
 class Torsion:
     name: str
-    a: ConnectedAtom
-    b: ConnectedAtom
-    c: ConnectedAtom
-    d: ConnectedAtom
+    a: UnresolvedAtom
+    b: UnresolvedAtom
+    c: UnresolvedAtom
+    d: UnresolvedAtom
 
 
 @attr.s(auto_attribs=True, frozen=True, slots=True)
-class Residue:
+class ChiSamples:
+    chi_dihedral: str
+    samples: Tuple[float, ...]
+    expansions: Tuple[float, ...]
+
+
+@attr.s(auto_attribs=True, frozen=True, slots=True)
+class SidechainBuilding:
+    chi_samples: ChiSamples
+
+
+@attr.s(auto_attribs=True, frozen=True, slots=True)
+class PolymerProperties:
+    is_polymer: bool
+    polymer_type: str
+    backbone_type: str
+    mainchain_atoms: Optional[Tuple[str, ...]]
+    sidechain_chirality: str
+    termini_variants: Tuple[str, ...]
+
+
+@attr.s(auto_attribs=True, frozen=True, slots=True)
+class ProtonationProperties:
+    protonated_atoms: Tuple[str, ...]
+    protonation_state: str
+    pH: float
```

Comment:
I assume this is to handle alternate protonations?   How does this work?

Reply:
I added this to future-proof the addition of alternate protonation states to the residue-type set so that the Packer (the PackerPalette) has a way to say "I don't want to include this protonated ASP residue type in this packing run." The PackerPalette has the code now to handle later additions of alternate protonation states. I don't, however, have any tests to make sure that logic is working correctly.